### DISCUSSION TITLE
🎉 Announcing a New Chapter for Ollama-webui: Help Us Choose Our New Name! 🚀

### DISCUSSION THREAD

**Discussion by tjbck** [open-webui/open-webui#602] [Announcements]
Hello, amazing ollama-webui community! 👋

First and foremost, we want to extend our heartfelt thanks to each and every one of you for your incredible support and enthusiasm. It's been an amazing journey so far with the ollama-webui. Your engagement and feedback have been pivotal in shaping this project.

### A New Direction: Name Change Announcement 🔄

However, with growth comes change. Recently, the Ollama team reached out to us with a valid concern: "We want to avoid confusion for our users. I think people are getting confused about Ollama-webui and Ollamahub as official Ollama trademarked items". In light of this, to avoid any confusion and maintain clarity for our users, we've decided it's time for a new name!

We're considering two potential names: **One-webui** and **Open-webui**. Here's the rationale behind each:

- **One-webui**: Symbolizes unity and the integration of various LLMs under one roof. It reflects our vision of being the singular, go-to platform for all your LLM needs.
- **Open-webui**: Emphasizes our commitment to openness and flexibility. It represents our dedication to supporting a broad range of LLMs, fostering an open community, and encouraging contributions.

### Exciting Roadmap Ahead: Expansion and Integration 🌟

Aligned with our name change, we're thrilled to share our future roadmap. While we remain committed to providing tight integration with Ollama, we're also expanding to directly utilise [llama.cpp](https://github.com/ggerganov/llama.cpp) and include first-class support for additional LLM runners like LiteLLM, vLLM and more. This evolution reflects our ambition to cater to a wider array of LLM applications and users.

We've been busy with our day jobs, but our passion for Ollama-webui (soon to be rebranded!) has never waned. We are dedicated to continuing to provide the best experience for running LLMs. Our goal is simple yet ambitious: to be the premier platform for LLM usage.

### We Need Your Help: Cast Your Vote! 🗳️

Now, we turn to you, our valued community, to help us choose our new name. What resonates more with you - **One-webui** or **Open-webui**? Your opinion matters immensely to us, and we're eager to hear your thoughts.

Please cast your vote, share your insights, or suggest alternative names if you have any brilliant ideas. Together, let's embark on this exciting new chapter!

With gratitude and excitement,

The Ollama-webui Team (soon to be renamed!)

---
# Poll: https://github.com/ollama-webui/ollama-webui/discussions/720
---

**Comment by 0xb15d3** [comment]
I'm fine with either names but FYI, I just googled to check if the name "ollama" has been trademarked, and seems like it's at its live/pending stage which means it's not fully trademarked yet afaik. Their claims of "trademarked items" seem a bit misleading ngl.

Having said that, moving away from ollama and integrating other LLM runners sound like a great plan. I mainly just use ollama-webui to interact with my vLLM server anyway, https://github.com/ollama/ollama/issues/2231 also raised a good point of ollama team not being very transparent with their roadmap/incorporating wanted features to ollama. Super excited for the future of this project, thank you for your hard work!

**Comment by shivaylamba** [comment]
I am fine with both names.

**Comment by nperez0111** [comment]
webui is a bit of an odd holdover. open-llm? llm-web?

**Comment by C00100011** [comment]
OpenLLMChat⭐️ Easy and explains a big aspect of this project.

**Comment by functorism** [comment]
I agree with @C00100011. The two provided names are too generic and offer little sticking power in people's minds. OpenLLMChat is probably the most on the nose, but utilitarian counter to the already chosen candidates. I also appreciate @nperez0111 take that maybe `webui` isn't necessary in the name for such a polished project.

I would encourage consideration of a bolder, more creative and inspiring name that can stick in people's minds - or - a name that very clearly describes the broader scope (as opposed to `ollama-webui`).

This project is an interaction portal of sorts. Names could range from all sorts, from "ChatWeaver", to "Portalis", etc, etc. Or more directly "OpenLLMChatBridge", etc.

Don't miss the chance to capitalize on a great name for a great project!

**Reply by C00100011** [reply]
I couldn't have said it better. 👌 Thanks @functorism

**Comment by justinh-rahb** [comment]
I agree with @functorism above.. this is a bit of a raw deal, we may as well make the most of it. We should create a brand that speaks to us. The poll choices aren't really doing it for me.

Either way leaves a sour taste in my mouth in regards to Ollama, bit of a reach complaining over a trademark they haven't secured yet (you moving out of the way would be in service of this btw). Let me be clear, until they do have a trademark you cannot be legally compelled to give up the domain you've already registered.. if they want you to do so *voluntarily* they should bear the cost.

I've been seeing and disregarding several very similar runners to Ollama, I'll be paying more attention now.

**Reply by 0xb15d3** [reply]
I agree with everything you said. If we really want to get into the semantics, the name "ollama" is derived from [LLaMA](https://en.wikipedia.org/wiki/LLaMA) and I personally don't think anyone has the right to claim the word. I wonder if ollama team will also go after https://github.com/ivanfioravanti/chatbot-ollama, https://github.com/ollama-ui/ollama-ui, https://github.com/hinterdupfinger/obsidian-ollama and https://github.com/ollama-interface/Ollama-Gui,  they all have their "trademarked"™️ name in their repo. 

Overall, I think it's a net positive for the community if we move away from ollama dependency and have direct llama.cpp integration instead. Also, for a suggestion, it would be great if I can just have one container that does everything by embedding llama.cpp to the project. It seems like it's on the roadmap  so I'm excited to see that feature in the future.

**Reply by functorism** [reply]
Supporting ollama is still a great feature, let's not this sour experience lead to bad product decisions. Supporting multiple model inference strategies is great, but avoiding ollama out of spite would be a bad outcome.

**Reply by justinh-rahb** [reply]
You're right of course @functorism and I don't want to see Ollama support removed or anything like that, but they're certainly not the only runtime out there. They have arguably been the most popular turnkey solution for GGUF models besides llama.cpp itself, but I wonder how long that will remain true.

**Comment by C00100011** [comment]
Just came with a random brain fart... 

How about "Llex" as in Large Language (Model) experience. Cause you get to experience the power of an LLM in a user friendly experience. 

Some random benefits include
- It is a available name in a lot of countries.
- Short and easy to remember.
- You can make easy alteration out of the name if it gets forked (like llex-connector, llex-importer, llexapp)
- It's catchy, what do you people think?

<img src="https://media0.giphy.com/media/d3mlE7uhX8KFgEmY/giphy.gif"/>

**Reply by justinh-rahb** [reply]
I like where you're heading, a unique name that can become a brand would be best. Riffing on yours a bit, I kinda like the acronym LMX.

**Comment by tjbck** [comment]
@nperez0111 @C00100011 @justinh-rahb  Absolutely! We appreciate your input. Let's take some time to explore and suggest creative names that truly resonate with the community. Feel free to share your ideas in the comments or submit them through our survey here: [Name Change Suggestions Survey](https://docs.google.com/forms/d/1-_J68B5xydRfSFHpYX83OUN5epBRwfe6sRGHDYy4Hqw/edit). Your voice matters, and we're excited to build a brand that reflects the spirit of our community. 🚀

**Reply by justinh-rahb** [reply]
Thanks are to you Timothy for being a good steward of this project, I think this is the best way to move forward given the circumstances. 

Ollama can have their name if they want it to themselves, we don't need it, we'll let the stars speak for us. 💫

**Comment by DennisPing** [comment]
I'd prefer a bolder name.

A real life example is [**oobabooga**](https://github.com/oobabooga/text-generation-webui). Everyone on the internet calls it "oobabooga" and **not** "text-generation-webui". Even if you misspell it badly, Google still presents it as a top result. Idk if they did any SEO hacks/tricks so I won't go there... oh, and future SEO is easier with a unique name :)

I took inspiration from another comment: Zooweaver

**Reply by justinh-rahb** [reply]
Not to my personal taste, but there's something to be said for boldness. I say we throw it in the ring and let the people decide!

**Comment by tjbck** [comment]
Loving all the suggestions so far, but Ideally we'd like to keep the name direct, simple and clean! "openllm-ui", "lmx-ui", "llex-ui" seem like great candidates to us at the moment, we were thinking of choosing between"one-webui" and "open-webui" for seo reasons so I believe we should also keep them as candidates! Personally leaning towards these options as once we get the ball rolling for direct llama.cpp integration, we will start including features that were only present in oobaboga like lora finetuning and more (but much more streamlined)! 

One big concern with using names like "openllm" is that there are already plethora of existing repos using the exact same name so we might want to avoid being too generic! Thanks for all the suggestions!

**Reply by KPHIBYE** [reply]
I would avoid the word `one` in the new name, since it is already heavily associated with Microsoft's `OneDrive` and `OneNote` and think that `openllm-ui` is a good name since it simultaneously communicates the open source nature, the relation to LLMs and the main purpose.

**Comment by Collected5353** [comment]
I wonder if we should consider branding in terms of how ollama-webui compares against the competitors. PrivateGPT is already out there, but a big differentiator for me with Webui has been the privacy and ease of deployment. 

Some ideas:

Private-WhispererWebUI emphasizes hiding secrets and providing unmatched privacy while offering powerful Reasoning, Alignment, and Generation functions.

**Comment by explorigin** [comment]
The trouble with changing the name is that we have a dang-cute llama. Whatever we change the name to, we need to have a dang-cute mascot to replace it. I suppose we could call it alpaca-something.

**Reply by explorigin** [reply]
AIpaca AI-paca (AI instead of AL)
<img width="295" alt="image" src="https://github.com/ollama-webui/ollama-webui/assets/697818/097c5e29-3695-437a-ab30-89ff5ea303f1">
(can't use that image, I just put it here to drum up support)

**Reply by explorigin** [reply]
Nevermind. There are already a lot of AIpaca things.

**Comment by isaacblum** [comment]
One-webui

**Reply by explorigin** [reply]
<img width="321" alt="image" src="https://github.com/ollama-webui/ollama-webui/assets/697818/0fcec3d3-8a41-43b4-8117-c9d2c1e4e8da">
https://fontawesome.com/icons/robot?f=classic&s=solid

I still prefer a furry animal. :-(

**Reply by tjbck** [reply]
![image](https://github.com/ollama-webui/ollama-webui/assets/25473318/e754bda1-26f7-4ebb-8df5-2591365a842c)

Unironically this could be our default logo.

**Reply by explorigin** [reply]
DogeAI! It has the novelty of oogabooga! I'm seriously in favor of this.

**Reply by justinh-rahb** [reply]
Hell why not, if Elon can name his AI Grok, this ain't the weirdest thing in the field.

**Reply by explorigin** [reply]
It'd be awesome if we could get a monochromatic lineart doge.

**Reply by tjbck** [reply]
For the name, we could call it D.O.G.3., Distributed Open Generative(AI) 3xperience. The more I think of it the more I'm inclined to choose this name over others.

**Reply by 0xb15d3** [reply]
I'm seriously in favor of this also! It's catchy and novel, I love it!

**Reply by tjbck** [reply]
@explorigin Here's what it looks like with the monochromatic line art.

![image](https://github.com/ollama-webui/ollama-webui/assets/25473318/86c5236c-9d05-455b-ab72-f5acfbb8a026)

![image](https://github.com/ollama-webui/ollama-webui/assets/25473318/33c2ed0e-479e-45fd-91df-8e1b844c9719)

**Reply by justinh-rahb** [reply]
I'm perfectly fine with this as a default option and the branding for the project, but I do think we'll need to make it possible for admins to customize, it won't be to everyone's taste.

**Reply by tjbck** [reply]
@justinh-rahb Indeed, already started working on it with #695 :)

**Reply by explorigin** [reply]
Second one is best. For the name: I think it should be something AI related. Doge-webui is too generic. We could go with the D. O. G. E. acronym but I don't understand the 3 instead of E and it's also less descriptive as an acronym (oogabooga is not actually named that in the repo so it helps with association). It's not like the Doge community so going to come back at us and ask us to change it.

This is why I recommended DogeAI. We get the Doge and it remains somewhat descriptive. We could just as easily call it DogeLLM.

**Reply by tjbck** [reply]
@explorigin All valid points! I was thinking of using the name `DOG3` because it would help us with the SEO (Nothing shows up when you google dog3 vs doge), also `dogeai` username seems to be taken for a Github organisation account unfortunately. I'm a bit hesitant to put LLM in the name as we're planning on expanding the webui to support various GenAI tasks like image generation and such, so having `AI` in our name instead I believe would be a more accurate description. 

Here are options that we're actively considering at the moment (other than open-webui, one-webui), We narrowed down the candidates based on the SEO results, and the availability of GitHub organisation usernames:

dog3 (Distributed Open GenAI 3xperience)
1. `dog3ai/ui`
2. `dog3ai/webui`
3. `dog3ai/dog3`

l3x (Local Large Language Model Experience)
4. `l3xai/ui`
5. `l3xai/webui`
6. `l3xai/l3x`

Which one do you guys prefer? feel free to suggest more repo names!

**Reply by explorigin** [reply]
The 3 in doge annoys me (which even I found odd, I'm rarely averse to 1337speak). Feelings are weird.

I vote for l3xai/ui even if we support multi-modal in the future.

**Reply by tjbck** [reply]
I suppose `l3xai/doge` can also be an option, but at that point `l3xai/ui` is more concise and more descriptive. Personally, fine with any of the options above!

**Comment by MikeHago** [comment]
OneOpenLLM-UI

**Comment by MikeHago** [comment]
To put it frankly and straightforwardly: It comes off as rather immature. It doesn't appear to be connected to AI, Ollama, or anything similar.

**Reply by 0xb15d3** [reply]
so is oobabooga, and many other AI products. Also, Ollama team has made it clear that they don't want to be associated with the project, I very much prefer the iconic name doge as it's very memorable.

**Reply by justinh-rahb** [reply]
> To put it frankly and straightforwardly: It comes off as rather immature. It doesn't appear to be connected to AI, Ollama, or anything similar.

My friend @MikeHago, please do explain what about a shiba inu is more objectionable than a llama or an alpaca or a vicuna or a capybara or a... do you see where this is going? Don't take life so seriously, it'll eat you up if you let it.

**Comment by MikeHago** [comment]
You're right, Justin, taking life too seriously gets us nowhere. But remember, our focus here is on brainstorming a name, not the broader strokes of life. Let’s take that step back and approach this with a fresh mindset. We aim to ignite creativity, not to follow the beaten path. Let's leave the 'animal farm' theme behind before we're pegged as a petting zoo for preschoolers. **Now's the chance to make Ollama WebUI stand out, showing it's not just another entry in the canine lineup.** This is the vision I wanted to put forward. Think on that. Give Ollama WebUI a name like a star in the AI universe.

**Comment by tjbck** [comment]
Opened a new poll here: https://github.com/ollama-webui/ollama-webui/discussions/720

Help us choose our new name!

**Comment by justinh-rahb** [comment]
I think all of the candidates are strong for their own reasons and representative of the community voice. May the best name win!

**Reply by adrianpunkt** [reply]
llm-ui or llm-webui ;)

**Comment by geekaz01d** [comment]
open-webui is the least informative name ever.

**Reply by justinh-rahb** [reply]
Thanks for you input, bit late.

**Reply by geekaz01d** [reply]
Yeah unfortunately I don't follow discussion threads on github so I missed this until the change was apparent. 

It's your decision to name things anything you want, but there is a 98% lower chance that anyone would have tried this project if it had not been associated to ollama in search results.

**Reply by justinh-rahb** [reply]
Ollama has a trademark and requested that we change the name, that wasn't an option.

**Reply by geekaz01d** [reply]
That's unfortunate. Oh well it's done!

**Comment by Darin755** [comment]
Open webUI is fairly vague and sounds like OpenAI which is kind of a nightmare for FOSS and privacy. Maybe sometime in the future you could come up with something better? I know naming is hard so I get it.
